client/views/6-AddNewContactView.js

```diff
@@ -0,0 +1,236 @@
+      console.log('newContact._id :', newContact._id);
+      console.log('In Redux : offlineContacts :', this.props.offlineContacts);
+
+      
+      if (this.props.offlineContacts.length > 0) {
+        console.log('ENTER');
+        // Update user's `contacts` DB, retrieve updated contacts objects and store in redux state
+        this.updateContactsInDB(this.props.offlineContacts);
+      }
+
+  }
+
+  updateContactsInDB = (contactsIdArray) => {
+    const payload = { 
+      id: this.props.myID,
+      contactsIdArray: contactsIdArray
+    };
+
+    fetch(`${BASE_URL}/user/add-contacts`, {
+      method: "POST",
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(payload)
+    })
+    .then(rawData => rawData.json())
+    .then( (updatedContacts) => {
+      // splice last item in the contacts array, as USERS actuall profile was pushed within `getUsersFriends` as last to avoid making double request
+      let myProfile = updatedContacts.splice(updatedContacts.length -1, 1)
+      this.props.storeMyProfile(myProfile[0]);
+      this.props.storeContacts(updatedContacts);
+      this.props.resetOfflineList();
+    })
+    .then(() => this.props.navigation.navigate('Dashboard'))
+  }
+
+
+  render() {
+    const { navigation } = this.props;
+    let newContact = navigation.getParam('newContact');
+
+    const { fName, lName, occupation, currentLocation: loc, email, birthplace  } = newContact.personal;
+    const { facebook, instagram, twitter, blog } = newContact.social;
+    const { linkedIn, github, cv, website } = newContact.networking;
+    const {photo} = newContact;
+    return (
+      
+      <View style={styles.container}>
+        <Button
+          icon={
+            <Icon
+              name='user'
+              size={15}
+              color='white'
+            />
+          }
+          title={`ADD ${fName.toUpperCase()}`}
+          onPress={ () => this.handleAddContact(newContact)}
+        />
+        <View style={styles.contactsWrapper}>
+          <Avatar
+            xlarge
+            rounded
+            source={{ uri: photo }}
+            onPress={() => console.log("Works!")}
+          />
+          <Text style={styles.name}> {`${fName} ${lName}`} </Text>
+          <Text style={styles.jobtitle}> {`${occupation} at Codeworks`} </Text>
+          <Text style={styles.details}>
+            Lives in - {`${loc.country}, ${loc.place}`}
+          </Text>
+          {/* <Text style={styles.details}>from {`${birthplace.country}`} </Text> */}
+
+          <Text 
+            style={styles.website}>
+            {`${website}`}
+          </Text>
+
+        </View>
+        <Text style={styles.category}> Social </Text>
+        <View style={styles.iconsContainer}>
+          <Icon
+            type='entypo'
+            name='facebook'
+            size={28}
+            raised={true}
+            onPress={() => canOpenURL(facebook)}
+          />
+
+          <Icon style={styles.icons}
```

Comment:
this is where having a separate Icon component that can just accept props would be handy as you used this component a lot in multiple other components.